### ISSUE TITLE
Build windows readme update

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaLang/julia#11602]
This should better be a pull request for the windows compile readme, but I don't really understand the issue well enough.
I'm building Julia with cygwin on windows and git decided to stop working with cygwin.
The error message was something like `git error: I don't work with the protocol https:\\`.
I thought, yeah no biggie, I just check out Julia with the windows cmd. This lead to the error `'\r': command not found`, or so. Fixes for this can be found here: [stackoverflow](http://stackoverflow.com/questions/11616835/r-command-not-found-bashrc-bash-profile)
But Julia still does not compile.
To really fix things, I needed to clone Julia with git from cygwin.  Probably `\r` wasn't fixed completely, or what else is platform dependent with git clone?! 
Fixing up git was a little magical. I tried a lot of solutions from the web, but what actually fixed it was to clone with `git@github.com:JuliaLang/julia.git` (which fails), but after that I tried again with the HTTPS URL, which then suddenly worked... I guess, trying to checkout via ssh somehow initialized the ssh store... or so...
Sorry for the fuzzy explanations, but I found these things out over the course of at least 1-2 months since I try not to waste much time with getting Julia to build.

As this is a cygwin issue, someone with better knowledge about this should update the Readme.

Best,
Simon

**Comment by tkelman** [comment]
Are you using Cygwin's git, or some copy that happens to be on your path? What does `which git` say?

**Comment by tkelman** [comment]
I see about 5 or 6 underspecified problems here, none of which you've given enough information about to know what to do with. Can you provide complete steps to reproduce any of these, a full error message, a proper bug report? Otherwise I'm just going to close this because it's nowhere near specific enough for me to know what to even write.

**Comment by SimonDanisch** [comment]
Sorry about this terribly unspecific issue, but I'm really happy that I was finally able to move on with this, and I can't easily reproduce this anymore... 
But it took my quite some time, so I hope this can be avoided for others. 
I had the hope, that someone knows the platform specific things that can happen if git clones from a different platform.

The broken (and now working) git is cygwin's: `/usr/bin/git`
After some googling, this seems to be the exact error `fatal: I don't handle protocol 'git clone https'`
As I haven't found a serious fix for this, I'm not sure what to do about it (and how to reproduce it)... but fixing this issue and cloning with cygwins git solves everything, so I thought that this is valuable information.

**Comment by tkelman** [comment]
It might be something having to do with needing curl or ca-certificates to be set up properly to work over https? I don't remember doing anything in particular to get Cygwin's git working properly, but my academic network might be way less restrictive than others.

Nothing here strikes me as "oh I've seen that before, you should do __." If you had seen a fork failure, I do have a workaround for that which should already be listed in the readme.

**Comment by SimonDanisch** [comment]
I can say now, that it's still not building because I didn't fix the `\r` error properly...As the real error is, that git is fetching the wrong line endings, I think it's no use to further investigate this.

I think it's a certificate error as well... And I really don't know why they keep popping up randomly. I had some issue with the certificates on msys2 as well and I'm pretty sure cygwin was working fine in the beginning. I use cygwin and msys2 only for building Julia, so I'd be surprised if I accidently changed anything.

**Comment by tkelman** [comment]
Cygwin's git should only ever clone with unix line endings. Are you trying to work inside a repo that was originally cloned via an msys or mingw git?

**Comment by SimonDanisch** [comment]
Okay, my error report must have been much worse than I thought.
That was the piece of information I wanted to relay.

I was trying to work in a repo, that I cloned with window's git, in order to get around cygwins broken git.
Which is, from what I learned, a bad idea (after working through some cryptic error messages and unrelated stack-overflow posts).... In between, I thought I fixed the line ending issue, but I guess it was only hidden in a long make error (which I can't really reproduce).

So, one should try to fix cygwins git, but the internet didn't give me great answers on how to fix it (user name and such was all set), and I kind of fixed it by accident in the end...
